Update WebKit tests for the unified HTTP stack

guoye-zhang opened the pull request:
#### 2cb0dc39f55ae7ae52d71ce6c7e6ba72143fad6c
<pre>
Update WebKit tests for the unified HTTP stack
<a href="https://bugs.webkit.org/show_bug.cgi?id=273806">https://bugs.webkit.org/show_bug.cgi?id=273806</a>

Reviewed by Alex Christensen and Alexey Proskuryakov.

Fixing 2 tests that fails on the unified HTTP stack.

* LayoutTests/imported/w3c/web-platform-tests/websockets/handlers/sleep_10_v13_wsh.py:
(web_socket_do_extra_handshake): Fixing imported/w3c/web-platform-tests/websockets/interfaces/WebSocket/close/close-connecting.html?wss, import upstream WPT changes.
* Tools/TestWebKitAPI/Tests/WebKitCocoa/Download.mm:
(TestWebKitAPI::ResumeWithExtraInitialDataOnDisk)): Fixing TestWebKitAPI.WKDownload.ResumeWithExtraInitialDataOnDisk, the new stack uses the actual file size on disk to determine where to resume the download.

Canonical link: <a href="https://commits.webkit.org/278771@main">https://commits.webkit.org/278771@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/79a6c179ee36b06221c3f59b66c1ff1bebf3cc10

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/50925 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/30224 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/3247 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/54183 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/1615 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/53227 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/36519 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/1272 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/41468 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/53023 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/27881 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/43872 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/22598 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/25214 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/1128 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/9365 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/47196 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/1198 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/55777 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/26031 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/1087 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/48879 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/27284 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/43944 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/47963 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/28162 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/7510 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/27015 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aproskuryakov commented on LayoutTests/http/wpt/fetch/resources/status-without-message.asis:
```diff
@@ -1 +1 @@
-HTTP/1.1 200
+HTTP/1.1 200 
```
Did the old test fail in Firefox and Chrome?

guoye-zhang replied:
We decided to change the h1 parser, so I'll revert this test change. This is not the only test that checks this behavior, and Chrome accepts it.

guoye-zhang committed:
Update WebKit tests for the unified HTTP stack
https://bugs.webkit.org/show_bug.cgi?id=273806

Reviewed by Alex Christensen and Alexey Proskuryakov.

Fixing 2 tests that fails on the unified HTTP stack.

* LayoutTests/imported/w3c/web-platform-tests/websockets/handlers/sleep_10_v13_wsh.py:
(web_socket_do_extra_handshake): Fixing imported/w3c/web-platform-tests/websockets/interfaces/WebSocket/close/close-connecting.html?wss, import upstream WPT changes.
* Tools/TestWebKitAPI/Tests/WebKitCocoa/Download.mm:
(TestWebKitAPI::ResumeWithExtraInitialDataOnDisk)): Fixing TestWebKitAPI.WKDownload.ResumeWithExtraInitialDataOnDisk, the new stack uses the actual file size on disk to determine where to resume the download.

Canonical link: https://commits.webkit.org/278771@main
